### PR TITLE
HIVE-27426:Upgrade kryo version in iceberg module

### PR DESCRIPTION
**Pull request by devaspatikrishnatri**
### What changes were proposed in this pull request?
Upgrade kryo version in iceberg module

### Why are the changes needed?
Fix Cves

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Locally built

### CODE REVIEW

**Comment by akshat0395** [comment]
@devaspatikrishnatri Can you please update the PR description with the relevant details and attach the output of mvn dependency:tree to the PR.

**Comment by devaspatikrishnatri** [comment]
[mvn_dependency_tree.txt](https://github.com/apache/hive/files/11955189/mvn_dependency_tree.txt)

File: iceberg/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <hive.path.to.root>..</hive.path.to.root>
     <path.to.iceberg.root>.</path.to.iceberg.root>
     <iceberg.version>1.3.0</iceberg.version>
-    <kryo-shaded.version>4.0.3</kryo-shaded.version>
+    <kryo.version>5.0.3</kryo.version>
```
Comment (akshat0395) [inline iceberg/pom.xml:29]: @devaspatikrishnatri I see in the attached dependency tree the kryo version is 5.2.0 `com.esotericsoftware:kryo:jar:5.2.0:compile`
While here i see its 5.0.3, is this property not taking effect?

**Comment by devaspatikrishnatri** [comment]
[mvn_dependency_tree.txt](https://github.com/apache/hive/files/11957242/mvn_dependency_tree.txt)
This is the dependency tree with iceberg module, the previous one was a generic dependency tree.
Here you can see the replacement.

**Comment by akshat0395** [comment]
> [mvn_dependency_tree.txt](https://github.com/apache/hive/files/11957242/mvn_dependency_tree.txt) This is the dependency tree with iceberg module, the previous one was a generic dependency tree. Here you can see the replacement.

@devaspatikrishnatri I see this has 5.2.0 as well

File: iceberg/pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@
...
     <iceberg.mockito-core.version>3.4.4</iceberg.mockito-core.version>
     <iceberg.avro.version>1.11.1</iceberg.avro.version>
     <iceberg.kryo.version>5.2.0</iceberg.kryo.version>
```
Comment (akshat0395) [inline iceberg/pom.xml:32]: @devaspatikrishnatri We already have iceberg.kryo.version here set to 5.2.0,  Why do we need another kryo.version with a lower value

**Review by akshat0395 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**